pyo3-derive-backend/src/pymethod.rs

```diff
@@ -291,24 +325,48 @@ pub(crate) fn impl_wrap_getter(cls: &syn::Type, spec: &FnSpec) -> syn::Result<To
     })
 }
 
-/// Generate functiona wrapper (PyCFunction, PyCFunctionWithKeywords)
-pub(crate) fn impl_wrap_setter(cls: &syn::Type, spec: &FnSpec<'_>) -> syn::Result<TokenStream> {
+fn impl_call_setter(spec: &FnSpec) -> syn::Result<TokenStream> {
+    let (py_arg, args) = split_off_python_arg(&spec.args);
+
+    if args.is_empty() {
+        return Err(syn::Error::new_spanned(
+            &spec.name,
+            "Setter function expected to have one argument",
+        ));
+    } else if args.len() > 1 {
+        return Err(syn::Error::new_spanned(
+            &args[1].ty,
+            "Setter function can have at most two arguments: one of pyo3::Python, and one other",
+        ));
+    }
+
     let name = &spec.name;
-    let python_name = &spec.python_name;
+    let fncall = if py_arg.is_some() {
+        quote!(pyo3::derive_utils::IntoPyResult::into_py_result(_slf.#name(_py, _val)))
+    } else {
+        quote!(pyo3::derive_utils::IntoPyResult::into_py_result(_slf.#name(_val)))
+    };
 
-    let val_ty = match &*spec.args {
-        [] => {
-            return Err(syn::Error::new_spanned(
-                &spec.name,
-                "Not enough arguments for setter {}::{}",
-            ))
+    Ok(fncall)
+}
+
+/// Generate functiona wrapper (PyCFunction, PyCFunctionWithKeywords)
+pub(crate) fn impl_wrap_setter(
+    cls: &syn::Type,
+    property_type: PropertyType,
+) -> syn::Result<TokenStream> {
+    let python_name;
```

Comment:
I recommend
```rust
let (python_name, setter_impl) = match property_type {
    ...
};
```

Reply:
👍 I did consider that style too, but marginally preferred the style I chose. I'll switch to what you propose.